src/metatensor/models/soap_bpnn/model.py

```diff
@@ -282,8 +282,14 @@ def forward(
         outputs: Dict[str, ModelOutput],
         selected_atoms: Optional[Labels] = None,
     ) -> Dict[str, TensorMap]:
-        if selected_atoms is not None:
-            raise NotImplementedError("SOAP-BPNN does not support selected atoms.")
+        for system in systems:
+            for species in system.species:
+                if species not in self.all_species:
+                    raise ValueError(
+                        "This model only supports systems with chemical "
+                        f"species {self.all_species}, but a system with species "
+                        f"{system.species} was provided"
+                    )
```

Comment:
This check is already done by MetatensorAtomisticModel if you specify the species in the model capabilities

Reply:
Hmmm ok, I would just like to make sure this is also enforced outside of that wrapper (e.g. when we do eval)

Reply:
You should be able to use `MetatensorAtomisticModel` for eval, isn't it?

Reply:
You're right, my bad